documentation/high-avail/AMD-Pensando_HA_Proposal.md

```diff
@@ -157,50 +222,141 @@ In this state the node is waiting for the Datapath to signal completion of state
-If the configured role is secondary the node goes to terminal state secondary. The DPU then waits for a switchover event to switch to primary. In this state the node keeps receiving flow sync messages from the peer and keeps the datapath ready for switchover.
+If the configured role is secondary the node goes to terminal state secondary. The DPU then waits for a switchover event to switch to primary. In this state the node keeps receiving flow sync messages from the peer and keeps the datapath ready for switchover. In this state switchover can occur due to 2 triggers
+1. Failure of the Primary
+2. Admin initiated switchover.
 
-**Primary**
+Failure of the Primary leads the Secondary to switchover to the Standalone-Primary State. When an administrative switchover is initiated the Secondary reaches the Standalone-Primary state via a sequence of states. This sequence helps reduce any drops in user traffic.
+
+**Switch Primary**
+This state is triggered on the secondary when there is a user trigger to drain traffic from the Primary and have the secondary take over. This user trigger itself is applied on the Primary by the admin and relayed to the secondary. In this state the DPU Dataplane is notified to switch to Primary state. When the DPU receives this notification the DP switches state to Primary and hence syncs any new flows and updates to the peer. At the same time it processes any flow updates from its peer. This is a short lived state where there is a 2 way sync between the peers. This is setup this way in preparation because during the subsequent states the Network advertisements are withdrawn from the old primary. While the network is converging traffic can be forwarded by the network to either the primary and the secondary. The 2 way sync between the peers handles this.
+
+**Wait Switch Primary**
+In this state the node waits for the DPU to complete switching the Dataplane to Active state and notify the State machine.
+
+**Switch Standalone**
+In this state the DPU is notified to switchover to standalone. The Primary (this node's peer) triggers this once the network convergence timeout has expired on the primary side. The Datapath switches to an intermediate state PRE_STANDALONE. In this state, datapath still accepts flow sync acknowledgments that were in flight from the peer. This state is needed for a duration of time so as to allow for draining all in flight messages.
+
+**Wait Switch Standalone**
+Once the DPU drain timer expires the DPU notifies switchover done which takes the state machine to exit this state and move to full standalone primary. A message is sent to the old primary to shutdown before exiting this state.
 
-This is the terminal state for the node when the node is configured as the primary. In this state the local datapath is forwarding traffic actively and synching state to the peer.
 
 ### SAI Definitions
 
 The SAI API calls necessary for communication between the SONIC stack and the SAI-DASH SDK are defined below.
 
```

Comment:
Could we add a section re: callback mechanism?  Needs to be defined so SyncD can register (ReDis is not used).
Should we use 2 different channels? Named pipe for data, another for all control messages?
We could do this, Sanjay and Guohan to look at examples.